sdk/communication/Azure.Communication.Administration/src/Models/InternalPhoneNumberReservationOperation.cs

```diff
@@ -0,0 +1,135 @@
+
+        /// <inheritdocs />
+        internal PhoneNumberReservation Value => OperationHelpers.GetValue(ref _value);
+
+        /// <inheritdocs />
+        internal bool HasCompleted => _hasCompleted;
+
+        /// <inheritdocs />
+        internal bool HasValue => (_rawResponse != null || _finalRawResponse != null) && _value != null;
+
+        /// <inheritdocs />
+        internal Response GetRawResponse() => HasCompleted ? _finalRawResponse : _rawResponse;
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        internal Response UpdateStatus(CancellationToken cancellationToken = default)
+            => UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        internal async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+            => await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <param name="async" />
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (HasCompleted)
+            {
+                return GetRawResponse();
+            }
+
+            if (cancellationToken == default)
+            {
+                cancellationToken = _cancellationToken;
+            }
+
+            Response<PhoneNumberReservation> update = async
+                ? await _client.GetReservationByIdAsync(reservationId: Id, cancellationToken: cancellationToken).ConfigureAwait(false)
+                : _client.GetReservationById(reservationId: Id, cancellationToken: cancellationToken);
+
+            var response = update.GetRawResponse();
+            _rawResponse = response;
+
+            if (IsOperationFinished(update))
+            {
+                lock (_lockObject)
+                {
+                    _finalRawResponse = response;
```

Comment:
I had a chat with Scott about this. Some of the LROs are not thread safe, but we decided to make ours threadsafe. Having said that, now that we have the lock, I do not think we need both `_final` and `_raw`. I originally suggested it to avoid `lock`. But there are some issues that convinced me it is cleaner to use `lock`. Also we are making network call, so a `lock` is a lot less costly anyways.